### PR TITLE
Bind exporter

### PR DESCRIPTION
**Pull request by tkummer33**
### What does this PR do?
Add role to deploy bind_exporter on target hosts. The role open's `"{{ bind_exporter_port }}"` in firewall. The stats page has to be available on target host. Prometheus.yml in setup-prometheus role was updated accordingly in PR #9. 

### How should this be tested?
Run the role and then check target host. The bind_exporter pod should be running.
Command below should return valid metrics.
```
curl <target_host>:{{ bind_exporter_port }}/metrics |grep bind
```

### Is there a relevant Issue open for this?
https://gitlab.consulting.redhat.com/rht-labs/labs-sre/documentation/-/issues/404

### Other Relevant info, PRs, etc.
https://github.com/redhat-cop/monitoring/pull/9

### People to notify
cc: @redhat-cop/monitoring

### CODE REVIEW

File: prometheus/generic/setup-bind-exporter/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+setup-node-exporter
+=========
+
+This role will instantiate a bind-exporter container on targeted hosts. The statistics page has to be enabled and available locally on targeted hosts.
```
Comment (pabrahamsson) [inline prometheus/generic/setup-bind-exporter/README.md]: Can you provide a link to describe how to do this?

File: prometheus/generic/setup-bind-exporter/tasks/prereqs.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    state: present
+  with_items:
+  - epel-release
+
+- name: Ensure pip is installed
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items:
+  - python-pip
```
Comment (pabrahamsson) [inline prometheus/generic/setup-bind-exporter/tasks/prereqs.yml]: Can these be combined into one task?

File: prometheus/generic/setup-bind-exporter/tasks/docker.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Run the bind_exporter
+  docker_container:
+    name: bind_exporter
+    image: "{{ bind_exporter_image }}:{{ bind_exporter_image_version }}"
+    pid_mode: "host"
```
Comment (pabrahamsson) [inline prometheus/generic/setup-bind-exporter/tasks/docker.yml]: Is this required? Do we need to share the host's process namespace?

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Thank you @tkummer33 
I've left a few comments inline

File: playbooks/infra-prometheus/setup-bind-exporter.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Install Docker
+  hosts: prometheus
+  become: True
+  roles:
+    - infra-ansible/roles/config-docker
```
Comment (pabrahamsson) [inline playbooks/infra-prometheus/setup-bind-exporter.yml:6]: Could you add a README to the `playbooks` folder with instructions on how to install the required roles?
I also noticed that the `requirements.yml` is using `master` for `infra-ansible`. We probably should pin this to the latest tag/release of `infra-ansible` instead.

File: prometheus/generic/setup-bind-exporter/README.md (hunk 1 above)
Comment (tkummer33) [reply prometheus/generic/setup-bind-exporter/README.md]: This should be resolved now. Added link to the docs and provided configuration snippet in the readme file

File: prometheus/generic/setup-bind-exporter/tasks/prereqs.yml (hunk 2 above)
Comment (tkummer33) [reply prometheus/generic/setup-bind-exporter/tasks/prereqs.yml]: I tried, but apparently python-pip is dependent on epel-release. Therefore the task fails "No package matching 'python-pip' found" if these are in single loop.

File: prometheus/generic/setup-bind-exporter/tasks/docker.yml (hunk 3 above)
Comment (tkummer33) [reply prometheus/generic/setup-bind-exporter/tasks/docker.yml]: Makes sense. Removed the pid_mode: line from the task

File: playbooks/infra-prometheus/setup-bind-exporter.yml (hunk 4 above)
Comment (tkummer33) [reply playbooks/infra-prometheus/setup-bind-exporter.yml:6]: Added the README to the directory. Changed the requirements.yml, to version: v1.0.16.
Seems that latest is not a valid tag on git repo, therefore I used the latest release.

File: prometheus/generic/setup-bind-exporter/tasks/prereqs.yml (hunk 2 above)
Comment (pabrahamsson) [reply prometheus/generic/setup-bind-exporter/tasks/prereqs.yml]: Thank you for looking into it, I should've verified this before asking.

File: prometheus/generic/setup-bind-exporter/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+setup-node-exporter
```
Comment (pabrahamsson) [inline prometheus/generic/setup-bind-exporter/README.md]: I missed this one the first time around, `s/node/bind/`

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Updates look good, found a small copy paste mistake I missed the first time. Once that's resolved we should be good.

**Review by pabrahamsson — APPROVED** [review]
LGTM

**Pull request merged**